### PR TITLE
Add filter to Widget Conditions to allow empty categories and tags as a dropdown option

### PR DESCRIPTION
**Pull request by jwenerd**
By default Widget Conditions does not display empty categories and tags. We get people setting up their sites with widgets and who want to setup their widget conditions prior to publishing posts on the site/category/tag.  This type of filter would make it quite easier enable. 

Without this type of filter, one need to add a filter to get_terms_args on all the requests which generate the dropdown ( like the following https://github.com/jwenerd/jetpack/compare/Automattic:master...jwenerd:master )

### CODE REVIEW

File: modules/widget-visibility/widget-conditions.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static function widget_conditions_options_echo( $major = '', $minor = ''
 				<option value=""><?php _e( 'All category pages', 'jetpack' ); ?></option>
 				<?php
 
-				$categories = get_categories( array( 'number' => 1000, 'orderby' => 'count', 'order' => 'DESC' ) );
+				$categories = get_categories( array( 'number' => 1000, 'orderby' => 'count', 'order' => 'DESC', 'hide_empty' => apply_filters( 'jetpack_widget_conditions_show_empty_terms' , false , $major , $minor ) ) );
```
Comment (dereksmart) [inline modules/widget-visibility/widget-conditions.php]: the filter name is confusing.  I'd expect `show_empty_terms` to want `true` in order to show.  `jetpack_widget_conditions_hide_empty_terms` is probably more fitting.

**Comment by dereksmart** [comment]
I think this is a good contribution.  I would find that annoying as well.  

I would just like to see a clearer filter name, and for it to be doc'd before it gets merged.

**Comment by samhotchkiss** [comment]
Hey @jwenerd , are you able to add in Derek's feedback above?  If you can get this in by Thursday, we should be able to get this into 3.7 which goes in to beta next week

Thanks!

**Comment by cfinke** [comment]
Is a filter necessary? The exclusion of empty categories was an oversight when the code was written initially, so I'd consider this a bug to be fixed rather than an enhancement.

**Comment by cfinke** [comment]
@jwenerd If you just drop the filter and change and it `hide_empty => false`, I could accept this pull request.

**Comment by samhotchkiss** [comment]
Closing until/unless we hear back from @jwenerd

**Comment by jwenerd** [comment]
My apologies!  My email filters were setup in the past in a way that I never got these messages... 

I've removed filter and set to `hide_empty => false` as @cfinke has suggested.  

The comparison is here: https://github.com/Automattic/jetpack/compare/master...jwenerd:master

**Comment by jeherve** [comment]
@jwenerd Could you [rebase your Pull Request](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) as to include the 2 commits on top of the most recent version of the Widget Visibility module code? We'll then review that PR once again and get it merged!

Thanks!

**Comment by dereksmart** [comment]
@jwenerd I've moved your commits to #5770 and will replace this branch.  Thank you!

**Pull request closed without merging**